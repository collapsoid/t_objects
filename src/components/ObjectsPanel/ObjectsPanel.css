.objects-panel {
  padding: 5px 10px;
}

.objects-panel__header-container {
  display: flex;
  justify-content: space-between;
  padding: 5px 0;
}

.objects-panel__objective {
  white-space: nowrap;
}

.objects-panel__content-container {
  display: grid;
  grid-template-columns: 240px auto;
  column-gap: 25px;
}