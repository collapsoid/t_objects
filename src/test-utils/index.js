import React from 'react';
import {createStore, combineReducers} from 'redux';
import {Provider} from 'react-redux';
import {render as rtlRender} from '@testing-library/react';

import * as reducers from '../store/reducers';

function render(
  ui,
  {
    initialState,
    store = createStore(combineReducers({...reducers}), initialState),
    ...renderOptions
  } = {}) {
  function Wrapper({children}) {
    return <Provider store={store}>{children}</Provider>;
  }

  return rtlRender(ui, {wrapper: Wrapper, ...renderOptions});
}

export * from '@testing-library/react';
export {render};