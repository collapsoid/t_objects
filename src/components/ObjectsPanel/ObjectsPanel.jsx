import React, {useState} from 'react';
import {connect} from 'react-redux';

import './ObjectsPanel.css';

import Objective from './Objective/Objective';
import HideButton from './HideButton/HideButton';
import ObjectsList from '../ObjectsList/ObjectsList';
import ObjectsCreationPanel from '../ObjectsCreationPanel/ObjectsCreationPanel';
import ObjectsReviewPanel from '../ObjectsReviewPanel/ObjectsReviewPanel';


const ObjectsPanel = ({currentTargetObjects}) => {
  const [isVisible, setIsVisible] = useState(true);
  const [editorState, setEditorState] = useState(null);

  return (
    <div className="objects-panel">
      <div className="objects-panel__header-container">
        <div className="objects-panel__objective">
          <Objective title="ДОБАВЬТЕ ЦЕЛЕВЫЕ ОБЪЕКТЫ И ОПРЕДЕЛИТЕ ИХ ПАРАМЕТРЫ" />
        </div>
        <div className="objects-panel__hide pseudo-btn" onClick={() => setIsVisible(!isVisible)}>
          <HideButton isVisible={isVisible} />
        </div>
      </div>
      {isVisible &&
        <div className="objects-panel__content-container">
          <div className="objects-panel__objects-list">
            <ObjectsList state={editorState} setState={setEditorState} />
          </div>
          {editorState &&
            <div className="objects-panel__objects-editor">
              {editorState === 'creation' && <ObjectsCreationPanel setEditorState={setEditorState} {...currentTargetObjects} />}
              {editorState === 'review' && <ObjectsReviewPanel setEditorState={setEditorState} {...currentTargetObjects} />}
            </div>
          }
        </div>
      }
    </div>
  );
};

const mapState = ({currentTargetObjects}) => ({currentTargetObjects});

export default connect(mapState)(ObjectsPanel);