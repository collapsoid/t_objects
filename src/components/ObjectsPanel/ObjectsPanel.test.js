import React from 'react';
import {fireEvent, render} from '../../test-utils';

import ObjectsPanel from './ObjectsPanel';

describe('ObjectsPanel', () => {
  test('should render the panel without editor', () => {
    const {asFragment} = render(<ObjectsPanel currentTargetObjects={null} />);
    const component = asFragment();

    expect(component).toMatchSnapshot();
  });

  describe('hide/unhide panel', () => {
    test('should render the hidden panel', () => {
      const {asFragment, getByText} = render(<ObjectsPanel currentTargetObjects={null} />);
  
      fireEvent.click(getByText(/свернуть/));
      const component = asFragment();
  
      expect(component).toMatchSnapshot();
    });
  
    test('should render the unhidden panel', () => {
      const {asFragment, getByText} = render(<ObjectsPanel currentTargetObjects={null} />);
  
      const firstRender = asFragment();
      fireEvent.click(getByText(/свернуть/));
      fireEvent.click(getByText(/развернуть/));
      const unhidden = asFragment();
  
      expect(unhidden).toEqual(firstRender);
    });
  });
});