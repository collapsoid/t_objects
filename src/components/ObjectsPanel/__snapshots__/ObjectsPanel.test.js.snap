// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ObjectsPanel hide/unhide panel should render the hidden panel 1`] = `
<DocumentFragment>
  <div
    class="objects-panel"
  >
    <div
      class="objects-panel__header-container"
    >
      <div
        class="objects-panel__objective"
      >
        <div
          class="objective"
        >
          <span
            class="objective__title"
          >
            ДОБАВЬТЕ ЦЕЛЕВЫЕ ОБЪЕКТЫ И ОПРЕДЕЛИТЕ ИХ ПАРАМЕТРЫ
          </span>
          <img
            alt="info"
            class="objective__ico"
            src="[object Object]"
          />
        </div>
      </div>
      <div
        class="objects-panel__hide pseudo-btn"
      >
        <div
          class="hide-btn"
        >
          <span
            class="hide-btn__text"
          >
            развернуть
          </span>
          <img
            alt="hide/unhide"
            class="hide-btn__ico"
            src="[object Object]"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`ObjectsPanel should render the panel without editor 1`] = `
<DocumentFragment>
  <div
    class="objects-panel"
  >
    <div
      class="objects-panel__header-container"
    >
      <div
        class="objects-panel__objective"
      >
        <div
          class="objective"
        >
          <span
            class="objective__title"
          >
            ДОБАВЬТЕ ЦЕЛЕВЫЕ ОБЪЕКТЫ И ОПРЕДЕЛИТЕ ИХ ПАРАМЕТРЫ
          </span>
          <img
            alt="info"
            class="objective__ico"
            src="[object Object]"
          />
        </div>
      </div>
      <div
        class="objects-panel__hide pseudo-btn"
      >
        <div
          class="hide-btn"
        >
          <span
            class="hide-btn__text"
          >
            свернуть
          </span>
          <img
            alt="hide/unhide"
            class="hide-btn__ico"
            src="[object Object]"
          />
        </div>
      </div>
    </div>
    <div
      class="objects-panel__content-container"
    >
      <div
        class="objects-panel__objects-list"
      >
        <div
          class="objects-list"
        >
          <ul
            class="objects-list__list"
          />
          <div
            class="objects-list__add-objects"
          >
            <button
              class="objects-list__add-objects-btn"
            >
              ДОБАВИТЬ ОБЪЕКТЫ
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
